### PR TITLE
Add simple admin authorization #8

### PR DESCRIPTION
**Pull request by MarkMarkyMarkus**

### CODE REVIEW

File: api/build.gradle.kts [hunk 1]
```diff
@@ -18,8 +19,8 @@ dependencies {
     testImplementation("org.springframework.boot:spring-boot-starter-test:2.2.5.RELEASE") {
         exclude(module = "junit")
     }
-    testImplementation("org.junit.jupiter:junit-jupiter-api:5.6.0")
-    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.6.0")
```
Comment (lamtev) [inline api/build.gradle.kts]: What's wrong with the lines? As I know, It's a good practice to separate implementation and runtime dependencies

File: api/build.gradle.kts [hunk 2]
```diff
@@ -28,3 +29,11 @@ dependencies {
...
     sourceCompatibility = JavaVersion.VERSION_13
 }
+
+tasks.withType<Test> {
+//    Enable JUnit5 support
+    useJUnitPlatform()
+    testLogging {
+        events("passed", "skipped", "failed")
+    }
+}
```
Comment (lamtev) [inline api/build.gradle.kts]: Please add a newline at end of file

File: api/src/main/java/org/accula/api/handlers/AdminHandler.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+@Component
+public class AdminHandler {
+    private static final String MESSAGE = "You are the admin now.";
+    private static final String ERROR = "Access denied. Try to login as ADMIN.";
+
+    @NotNull
+    public Mono<ServerResponse> getAdmin() {
+        return ok().bodyValue(MESSAGE)
+                // Unbelievable :)
+                .or(badRequest().bodyValue(ERROR));
```
Comment (lamtev) [inline api/src/main/java/org/accula/api/handlers/AdminHandler.java]: It's an unreachable branch, isn't it?

File: api/src/test/java/org/accula/api/routers/AdminRouterTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Test
+    @WithMockUser(username="admin",roles={"ADMIN"})
+    @DisplayName("Test correct admin request with ADMIN role")
+    public void testAdminRouterOk() {
+        client.get()
+                .uri("/admin")
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(String.class).isEqualTo("You are the admin now.");
```
Comment (lamtev) [inline api/src/test/java/org/accula/api/routers/AdminRouterTest.java]: I didn't review the whole PR for English. But here "an" is a better option than "the" I suppose

File: api/src/test/java/org/accula/api/routers/AdminRouterTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    @WithMockUser(username="user")
+    @DisplayName("Test invalid admin request with USER role")
+    public void testAdminRouterNotLoggedIn() {
+        client.get()
+                .uri("/admin")
+                .exchange()
+                .expectStatus().isForbidden()
+                .expectBody(String.class).isEqualTo("Access Denied");
+    }
+}
```
Comment (lamtev) [inline api/src/test/java/org/accula/api/routers/AdminRouterTest.java]: Please add a newline at end of file

File: api/build.gradle.kts (hunk 1 above)
Comment (MarkMarkyMarkus) [reply api/build.gradle.kts]: You are right, but I can't start JUnit5 tests with separate dependencies. Other tests use JUnit4.

File: api/src/main/java/org/accula/api/handlers/AdminHandler.java (hunk 3 above)
Comment (MarkMarkyMarkus) [reply api/src/main/java/org/accula/api/handlers/AdminHandler.java]: Yes, there is a comment.

File: api/src/test/java/org/accula/api/routers/AdminRouterTest.java (hunk 4 above)
Comment (MarkMarkyMarkus) [reply api/src/test/java/org/accula/api/routers/AdminRouterTest.java]: Maybe :)

File: api/build.gradle.kts (hunk 1 above)
Comment (lamtev) [reply api/build.gradle.kts]: Which ones use JUnit 4?

File: api/src/main/java/org/accula/api/handlers/AdminHandler.java (hunk 3 above)
Comment (lamtev) [reply api/src/main/java/org/accula/api/handlers/AdminHandler.java]: Comment sounds like "It's hard to believe, but this line is reachable" for me. So it doesn't matter

File: api/build.gradle.kts (hunk 1 above)
Comment (MarkMarkyMarkus) [reply api/build.gradle.kts]: GreetingRouterTest.java and StatusRouterTest.java, aren't they?

Comment (lamtev) [reply api/build.gradle.kts]: If so it's my fault. I was planning to use JUnit 5

Comment (lamtev) [reply api/build.gradle.kts]: I see, I mistakenly used junit4 TestRunner. I'll fix it soon. Thanks for pointing that out

**Pull request merged**